src/core/analytics.js

```diff
@@ -455,6 +467,33 @@ class Analytics {
     }
   }
 
+  /**
+   *
+   * @param {*} type
+   * @param {*} rudderElement
+   * @param {*} clientSuppliedIntegrations
+   * Sends cloud mode events to server
+   */
+  sendCloudModeEvents(type, rudderElement, clientSuppliedIntegrations) {
+    rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
+      this.integrationsData,
+      clientSuppliedIntegrations,
+    );
+
+    Object.setPrototypeOf(rudderElement, RudderElement.prototype);
```

Comment:
What's the need for this?

Reply:
remove this line in preProcessQueue as we need to add prototype of rudderElement so that it will have all the methods  of rudderElement class which is going to be use later